File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    int letters = 0, sent = 0, index = 0;
    float L, S;
    string s = get_string("Text: ");
    int a = strlen(s);
    //words
    int words = 1;
    for (int i = 0; i < a; i++)
    {
        if ((char) s[i] == (char) 32)
        {
            words += 1;
        }
    }
    //sentences
    for (int i = 0; i < a; i++)
    {
        if ((char) s[i] == (char) 33 || (char) s[i] == (char) 46 || (char) s[i] == (char) 63)
        {
            sent += 1;
        }
    }
    //letters
    for (int i = 0; i < a; i++)
    {
        if (((int) s[i] >= 65 && (int) s[i] <= 90) || ((int) s[i] >= 97 && (int) s[i] <= 122))
        {
            letters += 1;
        }
    }
    L = (float) letters / (float) words * 100;
    S = (float) sent / (float) words * 100;
    index = round(0.0588 * L - 0.296 * S - 15.8);


    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index <= 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    letter = sentece = 0
    word = 1
    length = len(text)
    # print(chr(53))
    for c in text:
        if ord(c) == 32:
            word += 1
        elif ord(c) == 33 or ord(c) == 46 or ord(c) == 63:
            sentece += 1
        elif (ord(c) >= 65 and ord(c) <= 90) or (ord(c) >= 97 and ord(c) <= 122):
            letter += 1
    L = round(letter / word * 100, 2)
    S = round(sentece / word * 100, 2)
    index = round(0.0588 * L - 0.296 * S - 15.8)
    if index > 16:
        print("Grade 16+")
    elif index <= 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


main()